packages/create-plugin/src/commands/generate.plopfile.ts

```diff
@@ -191,19 +217,30 @@ function getActionsForTemplateFolder({
   exportPath: string;
   templateData?: Record<string, string>;
 }) {
-  const files = glob.sync(`${folderPath}/**`, { dot: true });
+  let files = glob.sync(`${folderPath}/**`, { dot: true });
   function getExportFileName(f: string) {
     // yarn and npm packing will not include `.gitignore` files
     // so we have to manually rename them to add the dot prefix
     if (path.basename(f) === 'gitignore') {
       return '.gitignore';
     }
+    if (path.basename(f) === 'npmrc') {
+      return '.npmrc';
+    }
+
     return path.extname(f) === '.hbs' ? path.basename(f, '.hbs') : path.basename(f);
   }
   function getExportPath(f: string) {
     return path.relative(folderPath, path.dirname(f));
   }
 
+  // Remove the npmrc file if not running `pnpm`.
+  if (templateData.packageManagerName !== 'pnpm') {
```

Comment:
Nit: move this logic up to after line 220 for clarity

Reply:
Moved it up as suggested. 👍